Sources/Core/Transaction/CodableTransaction.swift

```diff
@@ -63,32 +63,32 @@ public struct CodableTransaction {
     // MARK: - Properties transaction type related either sends to a node if exist
 
     /// the nonce for the transaction
-    internal var nonce: BigUInt {
+    public var nonce: BigUInt {
         get { return envelope.nonce }
         set { envelope.nonce = newValue }
     }
 
     /// the max number of gas units allowed to process this transaction
-    internal var gasLimit: BigUInt {
+    public var gasLimit: BigUInt {
         get { return envelope.gasLimit }
         set { return envelope.gasLimit = newValue }
     }
 
     /// the price per gas unit for the tranaction (Legacy and EIP-2930 only)
-    internal var gasPrice: BigUInt? {
+    public var gasPrice: BigUInt? {
         get { return envelope.gasPrice }
         set { return envelope.gasPrice = newValue }
     }
 
     /// the max base fee per gas unit (EIP-1559 only)
     /// this value must be >= baseFee + maxPriorityFeePerGas
-    internal var maxFeePerGas: BigUInt? {
+    public var maxFeePerGas: BigUInt? {
         get { return envelope.maxFeePerGas }
         set { return envelope.maxFeePerGas = newValue }
     }
 
     /// the maximum tip to pay the miner (EIP-1559 only)
-    internal var maxPriorityFeePerGas: BigUInt? {
+    public var maxPriorityFeePerGas: BigUInt? {
```

Comment:
Making all these public it makes sense to me to drop policy related properties as well.

Like as we have external object responsible for gas/nonce resolving it's worth to add this one into transaction preparing pipeline, because it'll be just a internal change of `[Read|Write]Operation` logic, which is totally ok. 

Meanwhile we could update `Web3Provider` object to add their resolver property, to provide to user a way to setup resolver once for at least a session rather than stepping it for each TX.

And moving back to `[R|W]Operation` as it, as I remember, it also relays on `Web3Provider` in its logic it could read from that property as well, this would allow us to reduce entities duplications.

Intention of this proposal could be covered with messing words that I've used above, if so please don't hesitate to point me that out.

Reply:
* Policy related props now dropped (moved to separate file)
* Added an intermediate solution for adding `PolicyResolver`, might integrate differently later depending on what way we choose (see `TODO`s)
* I can def see making resolver a prop of `Web3Provier` wanted to get your opinion first though on the latest changes before going on (as this is getting big)

Let me know what you think of passing the policies like this. 

Another option which is more convenient could be to let the users set whatever they want to set manually on the tx e.g. gasLimit and nonce before hand and then this will be taken as manual policy and whatever's not specified as default/auto. Then there would be no need to pass in policies or only if required for some reason (although I wouldn't see the reason for now).

Reply:
I've leave a comment in that place. In shorts: current one is great, though I'm still thinking giving `resolver` property with some default implementation (means it isn't required to be set on provider instance init) is a good idea, but giving a user an optional ability to manage gas policies on exact call is required too.